tests/ui/const-generics/transmute.rs

```diff
@@ -67,10 +67,20 @@ fn transpose_with_const<const W: usize, const H: usize>(
   unsafe {
     std::mem::transmute(v)
   }
+
+}
+
+fn concat_tuple<const W: usize, const H: usize>(
+  v: ([u8; W], [u8; H]),
+) -> [u8; W + H] {
```

Comment:
is that even defined to have equivalent layout? :sweat_smile:

Reply:
https://doc.rust-lang.org/reference/type-layout.html#tuple-layout

Reply:
> Tuples are laid out according to the default representation.

That's `repr(Rust)`, which means their field order isn't defined. So you can do a concat if they're homogenous and have no padding but the order wouldn't be defined according to spec. The current layout algorithm doesn't touch homogenous structs but that's not guaranteed.

Reply:
Right, so what I'm saying is this could instead be implemented on ADTs, which are `#[repr(C)]` instead (and other constraints on alignment and padding). The ultimate point of this PR is to allow for concatenating arrays by making them contiguous in memory, then (possibly) transmuting them together, so it seems that going out of the way to make transmuting work is probably not the right path